packages/flutter/lib/src/rendering/sliver_fill.dart

```diff
@@ -83,21 +83,36 @@ class RenderSliverFillRemainingWithScrollable extends RenderSliverSingleBoxAdapt
     final SliverConstraints constraints = this.constraints;
     final double extent = constraints.remainingPaintExtent - math.min(constraints.overlap, 0.0);
 
+    final double cacheExtent = calculateCacheOffset(
+      constraints,
+      from: 0.0,
+      to: constraints.viewportMainAxisExtent,
```

Comment:
Should this be `contraints.remainingCacheExtent`?

Reply:
So `calculateCacheOffset` will use whichever is smaller - either the `constraints.viewportMainAxisExtent` or the `contraints.remainingCacheExtent`. The reason I use `constraints.viewportMainAxisExtent` is because that's the sliver's `scrollExtent`.

So the way I thought about it is that the `RenderSliverFillRemainingWithScrollable` widget has the max potential size of the `viewportMainAxisExtent`, even when part of it isn't visible and the child is the size of the remaining space in the viewport. So it's always taking up that much of the cacheExtent (or whatever is left of it). Also, `remainingCacheExtent` could be larger than the `viewportMainAxisExtent`, so it wouldn't make sense for it to use more cacheExtent than it's max potential size.

Does that make sense?

Reply:
Right! Yes, thank you for explaining that. I forget SliverFillRemaining is trickier then other slivers because it never actually extends into the cache extent, since it only fills the remaining space and no more than that. Gotcha. 👍